File: fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven minutesSoFar = expectedMinutesInOven - minutesSoFar

let preparationTimeInMinutes layerCount = layerCount * 2

let elapsedTimeInMinutes layerCount minutesSoFar = 
    preparationTimeInMinutes layerCount + minutesSoFar

File: csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;
 
    public int RemainingMinutesInOven(int previousMinutesInOven)
    {
        return ExpectedMinutesInOven() - previousMinutesInOven;
    }

    public int PreparationTimeInMinutes(int layers) => layers * 2;
    
    public int ElapsedTimeInMinutes(int layers, int previousMinutesInOven)
    {
        return PreparationTimeInMinutes(layers) + previousMinutesInOven;
    }
}
